semgrep_output_v1.atd

```diff
@@ -1730,6 +1730,10 @@ type subproject_stats = {
   dependency_sources: dependency_source_file list;
   (* Results of dependency resolution, empty if resolution failed *)
   ?resolved_stats: dependency_resolution_stats option;
+  (* Reason why resolution failed, empty if resolution succeeded *)
+  ?unresolved_reason: unresolved_reason option;
+  (* Errors encountered during subproject resolution *)
+  ?errors: sca_error list option;
```

Comment:
for list you can also use ~errors: sca_error list;
and it's backward compatible. It will just create an empty list by default when you look at the generated code.
The ocaml code (and probably Python code) is a little bit more convenient to use when you use ~.
@mjambon can maybe add more explanations about ~ but otherwise there is the ATD ref manual online.

Reply:
oh nice. i'll switch to that.!

Reply:
done in [688c5e7](https://github.com/semgrep/semgrep-interfaces/pull/356/commits/688c5e769de68e49e4f4bba60013e0011ca05ddf)